raitools/raitools/raianalyzer/raianalyzer.py

```diff
@@ -135,12 +156,85 @@ def list(self):
             configs[manager.name] = manager.list()
 
     def save(self, path):
-        # TODO: implement
-        raise NotImplementedError(
-            "Save not implemented for RAIAnalyzer")
+        """Save the RAIAnalyzer to the given path.
+
+        :param path: The path to save the RAIAnalyzer to.
+        :type path: str
+        """
+        # save each of the individual managers
+        for manager in self._managers:
+            manager.save(Path(path) / manager.name)
+        # save current state
+        self.train.to_pickle(Path(path) / TRAIN)
```

Comment:
We pickle data?

Reply:
I suppose that I was more expecting to see JSON. I don't have hugely strong opinions on the matter, though.

Reply:
hmm sure, I could change it to json.  I just saw here that pickle is faster, especially the newer pickle version (pickle-p2 in the graph), and it seemed to be the more recommended way just based on the number of answers recommending it:
https://stackoverflow.com/questions/17098654/how-to-store-a-dataframe-using-pandas
But it's probably better to save as json to be platform invariant.  I don't think this save/load method will be used for uploading the various artifacts anyway (at least for explanations it currently can't, since current UI code won't be able to handle it?).

Reply:
done, changed to json!